### PR TITLE
reuse http.client for each httpCaller

### PR DESCRIPTION
**Pull request by Nesze**

### CODE REVIEW

File: consumer/queue-caller.go [hunk 1]
```diff
@@ -113,3 +114,7 @@ func (p defaultHTTPCaller) DoReq(method, url string, body io.Reader, headers map
 
 	return ioutil.ReadAll(resp.Body)
 }
+
+func (caller defaultHTTPCaller) initializeClient() {
+	caller.client = http.Client{}
```
Comment (catalin-ilea) [inline consumer/queue-caller.go]: You should create the client only once and re-use it.

Comment (Nesze) [inline consumer/queue-caller.go]: Then there is no failover when one of the proxies is down.

Comment (catalin-ilea) [inline consumer/queue-caller.go]: So you're basically relying on client caching the dns lookup to make sure you always hit the same proxy-instance with create and consumer requests

Comment (Nesze) [inline consumer/queue-caller.go]: It's NOT a final solution as it's NOT yet clear how this works internally.

Comment (tosan88) [inline consumer/queue-caller.go]: give it a try ;)

**Comment by Nesze** [comment]
New approach: https://github.com/Financial-Times/message-queue-gonsumer/pull/9

**Pull request closed without merging**